blockprod/src/detail/block_maker.rs

```diff
@@ -155,16 +149,24 @@ impl BlockMaker {
             .await?
             .map_err(BlockProductionError::FailedConsensusInitialization)?;
 
-        block.update_consensus_data(consensus_data);
+        Ok(consensus_data)
+    }
 
+    pub fn solve_block(
+        chain_config: Arc<ChainConfig>,
+        mut block_header: BlockHeader,
+        current_tip_height: BlockHeight,
+        stop_flag: Arc<AtomicBool>,
+    ) -> Result<BlockHeader, BlockProductionError> {
         // TODO: use a separate executor for this loop to avoid starving tokio tasks
```

Comment:
I've used this PR to move `solve_block()` in a different thread, but I've done it inside `run()` instead of here in `solve_block()`. The reasoning is that since `run()` is `async`, we either need to wait to join in from `run()` meaning that `run()` won't be able to progress, or we spawn new threads per iteration, which is wrong.

So this comment can probably be dropped, but I'll do it in my PR.